packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+   * In case of an error, promise will reject with the error.
+   * @throws NOT_SUPPORTED_ON_PLATFORM Error if the DevicePermission.Media permission has not successfully granted.
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.request', permissions);
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    /**
+     * CameraImageProperties is for the image taken from the camera
+     * @beta
+     */
+    export interface CameraImageProperties extends VisualMediaProps {
+      /**
+       * The source in CameraImageProperties should always be CameraProps
+       */
+      sourceProps: CameraProps;
+
+      /**
+       * The common additional properties for image in mobile devices
+       */
+      commonImageprops: ImageProperties;
+    }
+
+    /**
+     * CameraImageProperties is for the image taken from the camera
+     * @beta
+     */
+    export interface GalleryImageProperties extends VisualMediaProps {
+      /**
+       * The source in GalleryImageProperties should always be GalleryProps
+       */
+      sourceProps: GalleryProps;
+
+      /**
+       * The common additional properties for image in mobile devices
+       */
+      commonImageprops: ImageProperties;
+    }
+
+    /**
+     *  All properties in ImageProperties are optional and have default values in the platform
+     *  Additional properties for image in mobile devices
+     * @beta
+     */
+    export interface ImageProperties {
+      /**
+       * Optional; indicate if inking on the selected Image is allowed or not
+       * Default value is false
+       */
+      shouldAllowInkingOnImages?: boolean;
+
+      /**
+       * Optional; indicate if putting text stickers on the selected Image is allowed or not
+       * Default value is false
+       */
+      shouldAllowTextStickersOnImages?: boolean;
+
+      /**
+       * Optional; indicate if the filter mode is enabled on the selected image
+       * Default value is false
+       */
+      enableFilterOnImages?: boolean;
+
+      /**
+       * Optional; Lets the developer specify the image output formats, more than one can be specified.
+       * Default value is Image.
+       */
+      imageFormatsAllowedToBeSelected?: ImageOutputFormats[];
+    }
+
+    /**
+     * Specifies the image output formats.
+     * @beta
+     */
+    export enum ImageOutputFormats {
+      /** Outputs image.  */
+      Image = 1,
+      /** Outputs pdf. */
+      PDF = 2,
+    }
+    const errorInvalidSource = 'Invalid input source.';
```

Comment:
If these errors are not going to change, please move them to `constants.ts` and import them here.

Reply:
This is also related with the `interface` comments.

Reply:
I see these errors are used in unit tests as well are we going to redefine everything again?

Reply:
I've moved those three constant strings into `constant.ts` to keep the style consistent in this codebase. 

However, I'd like to express my concerns about relocating those constants to the `constant.ts file`, and would like to hear your thoughts. 
The usage of those three constant string is limited, and they're closely tied to the visualMedia.image API. Ideally, the `constant.ts` file should only contain values shared across multiple APIs, preferably more than two, to prevent it from becoming bloated.

Regarding unit tests, as my current  knowledge in mobile coding conventions, we create new constants within the unit test module. While it might seem like we're redefining everything again, the advantage is that it can help detect inadvertent changes to constant strings in the code (eg. removed an empty space between words by mistake).

Reply:
I think it's a fair concern @MengyiGong . I would be fine with defining the constants in visualMedia.ts but outside of the visualMedia namespace. We could still export them and consume them in places like the unit tests then, but the constants wouldn't be exported for consumers of the library. Would that work for you?

Reply:
I'm ok with either way :) 
I would like to follow the current code style in the codebase, and if we have time, we can discuss about the refactor in the future.